src/ann_solo/spectrum_similarity.py

```diff
@@ -0,0 +1,878 @@
+    ssm : spectrum.SpectrumSpectrumMatch
+        The match between a query spectrum and a library spectrum.
+
+    Returns
+    -------
+    float
+        The Bray-Curtis distance between both spectra.
+    """
+    numerator = np.abs(
+        ssm.query_spectrum.intensity[ssm.peak_matches[:, 0]]
+        - ssm.library_spectrum.intensity[ssm.peak_matches[:, 1]]
+    ).sum()
+    denominator = (
+        ssm.query_spectrum.intensity[ssm.peak_matches[:, 0]]
+        + ssm.library_spectrum.intensity[ssm.peak_matches[:, 1]]
+    ).sum()
+    query_unique = ssm.query_spectrum.intensity[
+        np.setdiff1d(
+            np.arange(len(ssm.query_spectrum.intensity)),
+            ssm.peak_matches[:, 0],
+            assume_unique=True,
+        )
+    ].sum()
+    library_unique = ssm.library_spectrum.intensity[
+        np.setdiff1d(
+            np.arange(len(ssm.library_spectrum.intensity)),
+            ssm.peak_matches[:, 1],
+            assume_unique=True,
+        )
+    ].sum()
+    numerator += query_unique + library_unique
+    denominator += query_unique + library_unique
+    return numerator / denominator
+
+
+def canberra(ssm: spectrum.SpectrumSpectrumMatch) -> float:
+    """
+    Get the Canberra distance of peak matches between two spectra.
+
+    Parameters
+    ----------
+    ssm : spectrum.SpectrumSpectrumMatch
+        The match between a query spectrum and a library spectrum.
+
+    Returns
+    -------
+    float
+        The canberra distance of peak matches.
+    """
+    return scipy.spatial.distance.canberra(
+        ssm.query_spectrum.intensity[ssm.peak_matches[:, 0]],
+        ssm.library_spectrum.intensity[ssm.peak_matches[:, 1]],
+    )
+
+
+def jaccard(ssm: spectrum.SpectrumSpectrumMatch) -> float:
+    """
+    Get the Jaccard distance of peak matches between two spectra.
+
+    Parameters
+    ----------
+    ssm : spectrum.SpectrumSpectrumMatch
+        The match between a query spectrum and a library spectrum.
+
+    Returns
+    -------
+    float
+        The Jaccard distance of peak matches.
+    """
+    return scipy.spatial.distance.jaccard(
+        ssm.query_spectrum.intensity[ssm.peak_matches[:, 0]],
+        ssm.library_spectrum.intensity[ssm.peak_matches[:, 1]],
+    )
+
+
+def dice(ssm: spectrum.SpectrumSpectrumMatch) -> float:
+    """
+    Get the Dice dissimilarity of peak matches between two spectra.
+
+    Parameters
+    ----------
+    ssm : spectrum.SpectrumSpectrumMatch
+        The match between a query spectrum and a library spectrum.
+
+    Returns
+    -------
+    float
+        The Dice dissimilarity of peak matches.
+    """
+    return scipy.spatial.distance.dice(
+        ssm.query_spectrum.intensity[ssm.peak_matches[:, 0]],
+        ssm.library_spectrum.intensity[ssm.peak_matches[:, 1]],
+    )
+
+
+def improved_sim(ssm: spectrum.SpectrumSpectrumMatch) -> float:
+    """
+    Get the Improved Similarity Index of peak matches between two spectra.
+
+    TODO: reference needed.
```

Comment:
@issararab Do you have a (mass spec) reference where this "improved similarity" comes from?